gtime/plotting/preprocessing.py

```diff
@@ -0,0 +1,199 @@
+import pandas as pd
+import numpy as np
+from scipy.linalg import toeplitz
+
+
+def seasonal_split(df: pd.DataFrame, cycle: str = "year", freq=None, agg="mean"):
+    """
+    Converts time series to a DataFrame with columns for each ``cycle`` period.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+    cycle : str, cycle, calendar term ('year', 'quarter', 'month', 'week') or pandas offset string
+    freq : frequency, if specified, time series is resampled to it
+    agg : aggregation function used in resampling
+
+    Returns
+    -------
+    pd.DataFrame with seasonal columns
+
+    """
+
+    def _week_of_year(t: pd.Period):
```

Comment:
Functions in functions are not great, try to avoid them if possible

Reply:
fixed